File: src/main/java/com/devdaily/dbgrinder/model/FKDefinition.java
package com.devdaily.dbgrinder.model;

/**
  * FKDefinition
  * <p>A Container for a Foreign Key definition for a table</p>
  * @author  I.Holsman
  * @version 1.6
  * (c) Ian Holsman 1998
  * Released under GPL. See LICENSE for full details.
  */
import java.util.*;

public class FKDefinition
{
  protected String FKname;
  protected String FKtable;
  protected String PKtable;
  protected Vector FKfields;
  protected Vector PKfields;
  /**
    * Constructor
    */
  public FKDefinition( String pPKTable, String pFKTable, String pFKName )
  {
    FKtable = pFKTable;
    PKtable = pPKTable;
    if ( pFKName != null )
      FKname = pFKName;
    else
      FKname = FKtable + PKtable;
    FKfields = new Vector();
    PKfields = new Vector();
  }
  public  String getPKTableName( ) { return PKtable;}
  public  String getFKTableName( ) { return FKtable;}
  /**
   * add the corresponding Foreign Key Name and Primary Key name
   */
  public void addField( String FKFieldName, String PKFieldName )
    {
      FKfields.addElement( FKFieldName);
      PKfields.addElement( PKFieldName);
    }
  public String getFKColList() {
    String sRet = null;

    for (Enumeration e = FKfields.elements() ; e.hasMoreElements() ;) {
      if ( sRet != null )
        sRet = sRet + ", " + (String)e.nextElement();
      else
        sRet = (String)e.nextElement();
    }
    return sRet.toLowerCase();
  }
  public String getPKColList() {
    String sRet = null;

    for (Enumeration e = PKfields.elements() ; e.hasMoreElements() ;) {
      if ( sRet != null )
        sRet = sRet + ", " + (String)e.nextElement();
      else
        sRet = (String)e.nextElement();
    }
    return sRet.toLowerCase();
  }
  public String getPKName()
  {
    return (String)PKfields.firstElement();
  }
  public  String getFKName( )
  {
    return (String)FKfields.firstElement();
  }
  public Vector getFKFields()
  {
    return FKfields;
  }
  public Vector getPKFields()
  {
    return PKfields;
  }

}

File: src/main/scala/com/alvinalexander/cato/model/FKDefinition.scala
package com.alvinalexander.cato.model

/**
 *  (converted to Scala by Alvin J. Alexander)
  * FKDefinition
  * <p>A Container for a Foreign Key definition for a table</p>
  * @author  I.Holsman
  * @version 1.6
  * (c) Ian Holsman 1998
  * Released under GPL. See LICENSE for full details.
  */
import scala.collection.mutable.ArrayBuffer

// TODO this code needs to be seriously re-thought
case class FKDefinition(PKTable: String, FKTable: String, pFKName: String) {

    val FKname = if (pFKName != null) pFKName else FKTable + PKTable
    val FKfields = ArrayBuffer[String]()
    val PKfields = ArrayBuffer[String]()

    /**
     * add the corresponding Foreign Key Name and Primary Key name
     */
    def addField(FKFieldName: String, PKFieldName: String) {
        FKfields += FKFieldName
        PKfields += PKFieldName
    }

    def getFKColList: String = {
        var sRet = ""
        for (e <- FKfields) {
            // TODO i have no idea why sRet is set to null in the 'else' clause here
            if (sRet != null) sRet = sRet + ", " + e else sRet = e 
        }
        return sRet.toLowerCase
    }
    
    def getPKColList: String = {
        var sRet = ""
        for (e <- FKfields) {
            // TODO i have no idea why sRet is set to null in the 'else' clause here
            if (sRet != null) sRet + ", " + e else sRet = e 
        }
        return sRet.toLowerCase
    }

    // TODO these methods are dangerous
    def getPKName = PKfields(0)
    def getFKName = FKfields(0)

}



